src/meta/processors/partsMan/CreateSpaceProcessor.cpp

```diff
@@ -196,10 +233,29 @@ CreateSpaceProcessor::pickHostsWithZone(PartitionID partId,
             LOG(ERROR) << "Get zone " << zoneName << " failed";
             handleErrorCode(cpp2::ErrorCode::E_STORE_FAILURE);
             onFinished();
-            return pickedHosts;
+            return Status::Error(folly::stringPrintf("Get Zone %s failed", zoneName.c_str()));
         }
```

Comment:
kv of zone could be read at first, no need to check several times in loop of part and replica